tests/ssg_test_suite/test_env.py

```diff
@@ -228,25 +287,35 @@ def _save_state(self, state_name):
         state = self._create_new_image(self.current_container, state_name)
         return state
 
+    def get_ssh_port(self):
+        if self.domain_ip == 'localhost':
+            ports = self._get_container_ports(self.current_container)
+            if self.internal_ssh_port in ports:
+                ssh_port = ports[self.internal_ssh_port]
+        else:
+            ssh_port = self.internal_ssh_port
+        return ssh_port
+
+    def get_ssh_additional_options(self):
+        ssh_additional_options = super().get_ssh_additional_options()
+
+        port_opt = ['-o', 'Port={}'.format(self.ssh_port),]
+        for index, opt in enumerate(ssh_additional_options):
+            if opt.startswith('Port='):
+                ssh_additional_options[index] = port_opt[1]
+
+        if port_opt[1] not in ssh_additional_options:
```

Comment:
That's little bit more tricky - it assumes that there may be `Port=...` preceded by `-o`, in which case it replaces the `Port=...` with the right value.
If the port option is missing completely, it prepends both `-o` and `Port=...`. Filtering of the list is not so easy, because a subsequence would have to be taken out, and that's tricky to do in a simple for loop. What would be more understandable to you?

Reply:
@matejak  Aha, OK, this wasn't obvious for me. Could you add this explanation as a comment there?

Reply:
Why not - done.